salt/modules/zypperpkg.py

```diff
@@ -2487,9 +2566,18 @@ def search(criteria, refresh=False, **kwargs):
 
     out = {}
     for solvable in solvables:
-        out[solvable.getAttribute('name')] = dict()
+        pkg_name = solvable.getAttribute('name')
+        if pkg_name in out and not all_versions:
+            continue
+        elif pkg_name not in out:
+            out[pkg_name] = list()
+        pkg = dict()
+        if all_versions:
+            out[pkg_name].append(pkg)
+        else:
+            out[pkg_name] = pkg
```

Comment:
That's not good. If you `.get()` "name", then you are not sure if the key is there. That means, it might be `None`. Which you use then as a key in `out`...

Either use `solvable["name"]` and crash with `KeyError` or `if not bool(pkg_name): continue`.